shell/plugins/dashboard-store/__tests__/actions.spec.js

```diff
@@ -0,0 +1,125 @@
+import _actions from '@shell/plugins/dashboard-store/actions';
+
+const { findAll } = _actions;
+
+describe('dashboard-store: actions', () => {
+  let dispatchCalls = {};
+  let commitCalls = {};
+  // we're dumping our commits into a map so we can reference them
+  // actions and mutations are stored against their name in an array so we can capture if they're called multiple times
+  // arguments past the name of the action or mutation are stored in an array in case there are multiple
+  const commit = (mutation, ...args) => {
+    commitCalls[mutation] = [...(commitCalls[mutation] || []), args];
+  };
+  const dispatch = (action, ...args) => {
+    dispatchCalls[action] = [...(dispatchCalls[action] || []), args];
+    if (action === 'request') {
+      return { data: [] };
+    }
+  };
+  const state = { config: { namespace: 'unitTest' } };
+  const getters = {
+    normalizeType:    (type) => type,
+    typeRegistered:   (type) => type === 'foo',
+    haveAll:          (type) => type === 'baz',
+    haveAllNamespace: (type, namespaced) => type === 'far' && namespaced === 'faz',
+    all:              (type) => [type],
+    urlFor:           (type, _, opt) => type, // we're not testing the urlFor getter so we don't need to do anything with opt here
+  };
+  const rootGetters = {
+    'type-map/optionsFor': (type) => {
+      return {};
+    },
+    'auth/fromHeader': 'foo'
+  };
+
+  // we're not testing function output based off of state or getter inputs here since they are dependencies and should be tested independently
+  const ctx = {
+    state,
+    getters,
+    rootGetters,
+    commit,
+    dispatch
+  };
+
+  // setting afterEach here seems reasonable but isn't allowed by the linter
+  const resetCalls = () => {
+    dispatchCalls = {};
+    commitCalls = {};
+  };
+
+  describe('dashboard-store > actions > findAll', () => {
+    let findAllPromise;
+
+    it('expects findAll to return a promise', () => {
+      resetCalls();
+      findAllPromise = findAll(ctx, { type: 'type:foo' });
+
+      expect(typeof findAllPromise.then).toBe('function');
+    });
+
+    it('expects basic findAll to return an array asynchronously', async() => {
+      const findAllReturnValue = await findAllPromise;
```

Comment:
Where is `findAllPromise` supposed to come from? If this has to be defined outside, we should create a `describe` wrapper where we want to collect all the cases to be analyzed in separated tests as you did.

Reply:
findAllPromise is defined just a few lines above this one and is the un-awaited return value from findAll which is an async function and thus returns a promise. There is a test for that as well.

This has since been updated but the general behavior of the test is the same, we need to test for both the synchronous return value of the async function (which should be a promise) and the async return value after the await keyword in order to ensure that our tests are deterministic and point to the correct type of failure.